### PR TITLE
fix: GRD-77625 Mongo Atlas Input Plugin.

### PR DESCRIPTION
**Pull request by chirag-ibm**
[GRD-77625](https://jira-ds.rtp.raleigh.ibm.com:8443/browse/GRD-77625)

This PR includes fixes in Mongo Atlas input plugin.

### CODE REVIEW

**Review by apurva-birajdar — APPROVED** [review]

**Review by PratikshaSonawane — APPROVED** [review]

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 1]
```diff
@@ -7,72 +7,127 @@
     * Guardium Data Protection: 11.4 and above
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:7]: 11.4 and later.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 2]
```diff
@@ -7,72 +7,127 @@
...
+universal connector that is featured in IBM Security Guardium. It reads events 
+and messages from the Mongo Atlas audit log 
+into a [Guardium record](https://github.com/IBM/universal-connectors/blob/main/common/src/main/java/com/ibm/guardium/universalconnector/commons/structures/Record.java) instance 
+(which is a standard structure made out of several parts). The information is then 
+sent over to Guardium. Guardium records include the accessor (the person who tried 
+to access the data), the session, data, and exceptions. If there are no errors, the 
+data contains details about the query "construct". The construct details the main 
+action (verb) and collections (objects) involved.
 
 In order to support a few features one zip has to be added with the name "guardium_logstash-offline-plugins-mongo-atlas.zip".
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:20]: To support Universal Connector features for MongoDB Atlas, add the ZIP file *guardium_logstash-offline-plugins-mongo-atlas.zip*.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 3]
```diff
@@ -7,72 +7,127 @@
...
-  4. Select Dedicated Cluster.
-  5. Select your preferred Cloud Provider & Region
-  6. Select your preferred Cluster Tier.
-  7. Enter a name for your cluster in the Cluster Name field.
-  8. Click Create Cluster to deploy the cluster.
-  Now that your cluster is provisioned.
-  For more information https://www.mongodb.com/docs/atlas/tutorial/create-new-cluster/.
+1. Login to Atlas using https://cloud.mongodb.com/.
+2. Click 'Build a cluster'.
+3. If 'Build a cluster' option is unavailable,
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:25]: If 'Build a cluster' is not available, you need to create and provision a new cluster, as follows:

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 4]
```diff
@@ -7,72 +7,127 @@
...
+   Select **Database** from the menu at left and
+   Select the **Create** option in the top right corner.
+4. Select Dedicated Cluster.
+5. Select your preferred Cloud Provider & Region
+6. Select your preferred Cluster Tier.
+7. Enter a name for your cluster in the Cluster Name field.
+8. Click Create Cluster to deploy the cluster.
+   Now that your cluster is provisioned.
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:33]: Your cluster is now provisioned.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 5]
```diff
@@ -7,72 +7,127 @@
...
+9. Click **Next**.
+10. Copy and save the **Public Key**.
+11. Copy and save the **Private Key**.
+12. Click **Add Access list Entry**.
+    1. Enter an IP address from which you want Atlas to
+       accept API requests for this API Key. You can add the current IP address
+       just by clicking **Use Current IP address**.
+    2. Click **Save**.
+13. Click **Done**.
+14. Select the project from the top left corner
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:66]: Select the Project from the list below *Organizations*.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 6]
```diff
@@ -7,72 +7,127 @@
...
+    list below to the Organizations list.
+15. In the Security section of the left navigation, click on **Network Access**.
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:68]: In the Security section, click **Network Access**.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 7]
```diff
@@ -7,72 +7,127 @@
...
-  3. Paste this text and click Save
-  {
-     "atype": {
-      "$in": [
-       "authCheck",
-       "authenticate"
-      ]
+1. In the **Security** section of the left navigation, click **Advanced**.
+2. Click **Edit** just below to **Database Auditing** button.
+3. Paste below text and click **Save**.
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:84]: Paste the following code and then click **Save**.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 8]
```diff
@@ -95,19 +150,130 @@ If no traffic is observed and the API key configured properly, revalidate the IP
...
+* **group-id**: GroupId is equivalent to the project ID. To find it,
+    1. Go to your Organization.
+    2. Select **Projects** from the left menu.
+    3. Select the required project.
+    4. Click on the three dots right by the name of the selected project in the
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:166]: Click the ellipses (...) next to the name of the selected project

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 9]
```diff
@@ -7,72 +7,127 @@
...
-  10. Click on 'ADD IP ADDRESS' button.
-  11. Add IP address and and click on 'Confirm'.
-  For more information, https://www.mongodb.com/docs/atlas/configure-api-access/#add-an-api-access-list-entry.
-
-**Note**:
-If no traffic is observed and the API key configured properly, revalidate the IP in the allowed access list by removing and adding it again, and recreate the UC connection. 
+1. Navigate to the Access Manager page for your organization.
+2. Select **Projects** from the menu at left.
+3. Select a Project from the list.
+4. Click on three dots adjacent to the Project selection menu
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:49]: Click the ellipses (...) next to the Project menu (in the top left corner below the Organization menu).

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 10]
```diff
@@ -95,19 +150,130 @@ If no traffic is observed and the API key configured properly, revalidate the IP
...
+    3. Select the required project.
+    4. Click on the three dots right by the name of the selected project in the
+       top left corner.
+    5. Select **Project Settings**.
+    6. Copy the **Project ID** displayed on the page.
+
+* **hostname**: Hostname is equivalent to the cluster hostname. To find it,
+    1. Click **All Clusters** in the top right corner.
+    2. Select the cluster from the list.
+    3. There will be a panel named **Region**.
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:174]: From the **Region** panel, select any shard. After you select the shard, the **Status** page displays.

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 11]
```diff
@@ -95,19 +150,130 @@ If no traffic is observed and the API key configured properly, revalidate the IP
...
+    4. Select any shard among the listed inside the region panel.
+    5. You will land to the **Status** page.
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:176]: 6. On the *Status* page, the page title includes <hostname>:<port>. The host name format is:

   ```
   <cluster-name>-<shard-seq-seq>.<identifier>.mongodb.net
   ```
   For example: 
   ```
   cluster2-shard-00-01.i2jq9.mongodb.net
   ```

File: input-plugin/logstash-input-mongo-atlas/README.md [hunk 12]
```diff
@@ -95,19 +150,130 @@ If no traffic is observed and the API key configured properly, revalidate the IP
...
+    6. The label of the title of the page will be in the format of <hostname>:<port>.
+    7. Take the host name. It will be in the format of,
+       `<cluster-name>-<shard-seq-seq>.<identifier>.mongodb.net`. For example,
+       `cluster2-shard-00-01.i2jq9.mongodb.net`
 
+### Example
+#### Mongo  event
```
Comment (mllezak) [inline input-plugin/logstash-input-mongo-atlas/README.md:183]: #### Mongo event

**Review by mllezak — CHANGES_REQUESTED** [review]
I made a bunch of small updates -- please let me know if you have questions.

**Review by mllezak — COMMENTED** [review]
And -- there are a few sections where I can't edit (or leave a note): 


Line 81/135 (under Authorizing outgoing traffic from Mongo Atlas:

Change: 
 2. Issue these command:
    • grdapi add_domain_to_universal_connector_allowed_domains domain=cloud.mongodb.com

To: 
 2. Issue the following command:
    ```
   grdapi add_domain_to_universal_connector_allowed_domains domain=cloud.mongodb.com
   ```
Line 91/145

Change step 8 to:

Make sure that the `type` filed matches in both the input and filter configuration sectiosn. This field must be unique for every individual connector that you add.

Supported audit messages & commands notes (lines 120/287 to 122/289:

**Notes:**
* To make sure that events are handled properly, take the following steps:
   *  Set MongoDB access control, because messages with no users are removed.
   * Do not filter `authcheck` and `authenticate` events out of the MongoDB audit log messages.

Under Limitations (lines 136/303):

* `Client Host Name` is not supported. For system-generated queries, 'Server Host Name' and 'Client Host Name' are the same.
* IPv6 addresses are typically supported by the MongoDB and filter plug-ins. However, IPV6 is not fully supported by the Guardium pipeline.
* `Source Program` is left blank.
* Mentioning 'Audit filter criteria on MongoDB' captures all of the events. Set the audit filter criteria as needed to avoid unnecessary logs.


**** 
I'm fine after these changes are made.

**Pull request merged**